src/oc/lib/api/common.clj

```diff
@@ -187,17 +187,49 @@
    If no Authorization headers are supplied return nil."
   [headers passphrase]
   (when-let [token (get-token headers)]
-    (if (jwt/valid? token passphrase)
-      {:jwtoken token
-       :user (:claims (jwt/decode token))}
-      {:jwtoken false})))
+    (cond
+
+     (jwt/valid? token passphrase)
+     {:jwtoken token
+      :user (:claims (jwt/decode token))}
+
+     ;; identity token
+     (and (jwt/check-token token passphrase)
+          (:id-token (:claims (jwt/decode token))))
+     {:jwtoken false :id-token token}
+
+     :default
+     {:jwtoken false})))
+
+(defn allow-id-token
+  "Allow options request. Allow jwtoken. Allow id token. Allow anonymous."
+  [ctx]
+  (cond
+
+   (= (-> ctx :request :request-method) :options)
+   true
+
+   (:jwtoken ctx)
+   true
```

Comment:
I think if you want to allow users with the JWT you should return `(allow-authenticated ctx)` or i will be able to see stuff that is allowed for id-token with a fake JWT. Right?

Reply:
The existing code doesn't do this. You should look at `allow-anonymous` and `allow-authenticated`. You will see that the jwtoken is decoded and validated when initializing context.